### DISCUSSION TITLE
GOPROXY credentials showing up in debug logs

### DISCUSSION THREAD

**Discussion by Keysox** [renovatebot/renovate#13690] [Request Help]
We are using a self-hosted instance of Renovate and are setting an environment variable similar to:
`GOPROXY="https://${USERNAME}:${PASSWORD}@example.com"`

This seems to be working except our secure creds are leaking due to this line: https://github.com/renovatebot/renovate/blob/ec723e3e076f46424fe5d4fd34ba12e5cce18625/lib/workers/repository/stats.ts#L68

Is there another way to authenticate or should I create an issue for this?

**Comment by rarkins** [comment]
If you duplicate that secret also to renovate's "secrets" object then you might find its sanitized as a result. Right now it doesn't know that secret to sanitize it

**Reply by Keysox** [reply]
I had forgotten that field existed, thanks for the information! I wasn't able to get it to work properly but may have been doing something wrong.

We ended up refactoring it to use a `.netrc` file to authenticate instead
